backend/homework/9/src/main/java/com/example/demo/controller/VehicleController.java

```diff
@@ -0,0 +1,89 @@
+package com.example.demo.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import com.example.demo.dto.VehicleDTO;
+import com.example.demo.entity.Vehicle;
+import com.example.demo.service.VehicleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.util.List;
+
+/**
+ * Controller class responsible for handling HTTP requests related to vehicles.
+ */
+@RestController
+@RequestMapping("/vehicles")
+public class VehicleController {
+
+    private final VehicleService vehicleService;
+
+    /**
+     * Constructor for VehicleController.
+     *
+     * @param vehicleService The service responsible for handling vehicle-related operations.
+     */
+    @Autowired
+    public VehicleController(VehicleService vehicleService) {
+        this.vehicleService = vehicleService;
+    }
+
+    /**
+     * Handles GET requests to retrieve a list of all vehicles.
+     *
+     * @return ResponseEntity containing a list of VehicleDTOs.
+     */
+    @GetMapping
+    public ResponseEntity<List<VehicleDTO>> getAllVehicles() {
+        List<VehicleDTO> vehicles = vehicleService.getAllVehicles();
```

Comment:
Let the service function return ResponseEntity, Incase the function returns an exception the response can still be 200 ok